projects/core/src/lib/classes/adapters/content-editable-element.ts

```diff
@@ -0,0 +1,110 @@
+type IfEquals<X, Y, A = X, B = never> =
+    (<T>() => T extends X ? 1 : 2) extends <T>() => T extends Y ? 1 : 2 ? A : B;
+
+type WritableKeys<T> = {
+    [P in keyof T]-?: IfEquals<{[Q in P]: T[P]}, {-readonly [Q in P]: T[P]}, P>;
+}[keyof T];
+
+function isWritable<T>(obj: T, property: keyof T): property is WritableKeys<T> {
+    const descriptor = Object.getOwnPropertyDescriptor(obj, property);
+
+    return Boolean(descriptor && descriptor.writable);
+}
+
+export class MaskitoReactContentEditableElement {
+    private host: HTMLElement;
+    proxy: HTMLInputElement;
+
+    constructor(host: HTMLElement) {
+        if (!host.isContentEditable) {
+            throw new Error(
+                'Passed into MaskitoReactContentEditableElement element should be contenteditable',
+            );
+        }
+
+        this.host = host;
+
+        // Proxy used to forward properties from HTMLElement to our instance
+        this.proxy = new Proxy(this, {
```

Comment:
i didn't found any possible ways to automaticly forward all properties to host element except declared in MaskitoReactContentEditableElement, so i just added Proxy instance for this purpose